### DISCUSSION TITLE
Enhance plugin - Option to add @SuppressFBWarnings annotation

### DISCUSSION THREAD

**Discussion by NicklasWallgren** [hibernate/hibernate-orm#5668] [Design Proposals]
Lombok has an option to add `@SuppressFBWarnings` to generated code via `lombok.extern.findbugs.addSuppressFBWarnings`
It would be really helpful if the hibernate `gradle`/`maven` enhance plugins would provide a similar option.

We're currently struggling with warnings related to hibernate generated code in our entities, in SpotBugs.

```
NP | Non-null field $_hibernate_attributeInterceptor is not initialized by new <entity>
NP | Non-null field $_hibernate_entityEntryHolder is not initialized by new <entity>
NP | Non-null field $_hibernate_nextManagedEntity is not initialized by new <entity>
NP | Non-null field $_hibernate_previousManagedEntity is not initialized by new <entity>
NP | Non-null field $_hibernate_tracker is not initialized by new <entity>
```

**Comment by Sanne** [comment]
Sure, sounds reasonable :)  Will you try a patch?

**Reply by NicklasWallgren** [reply]
Great, I'm not to familiar with code base, but I took a first stab at it.
https://github.com/hibernate/hibernate-orm/compare/main...NicklasWallgren:hibernate-orm:option-to-add-suppress-fb-warnings

Am I on the right track? 🙂

**Reply by Sanne** [reply]
I think so, yes!

**Reply by NicklasWallgren** [reply]
Great, thanks.

I opened a PR, https://github.com/hibernate/hibernate-orm/pull/5790

**Reply by NicklasWallgren** [reply]
It would also be nice to add the `jakarta.annotation.Generated` annotation, so that tools like Jacoco can exclude hibernate-generated fields and methods from coverage reports.

**Reply by NicklasWallgren** [reply]
@Sanne How should we processed with this issue, should I update the PR?

**Comment by sebersole** [comment]
We discussed this as a team and the consensus was that we are not really keen on the idea of adding support for particular annotations one-by-one.  We believe this is a use case that can be generalized.  

Let's discuss approaches for that here or on Zulip
